### ISSUE TITLE
Infinite recursion bug

### ISSUE THREAD

**Issue by russkel** [adamrehn/ue4cli#47]
Using master branch I am getting these issues, my Unreal Engine lives in ~/UnrealEngine and is 4.27.

```
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 632, in _getEngineVersionDetails
    versionFile = os.path.join(self.getEngineRoot(), 'Engine', 'Build', 'Build.version')
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 60, in getEngineRoot
    self._engineRoot = self._getEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 620, in _getEngineRoot
    return self._detectEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerLinux.py", line 15, in _detectEngineRoot
    if self._getEngineVersionDetails()['MajorVersion'] >= 5:
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 632, in _getEngineVersionDetails
    versionFile = os.path.join(self.getEngineRoot(), 'Engine', 'Build', 'Build.version')
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 60, in getEngineRoot
    self._engineRoot = self._getEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 620, in _getEngineRoot
    return self._detectEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerLinux.py", line 15, in _detectEngineRoot
    if self._getEngineVersionDetails()['MajorVersion'] >= 5:
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 632, in _getEngineVersionDetails
    versionFile = os.path.join(self.getEngineRoot(), 'Engine', 'Build', 'Build.version')
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 60, in getEngineRoot
    self._engineRoot = self._getEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 620, in _getEngineRoot
    return self._detectEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerLinux.py", line 15, in _detectEngineRoot
    if self._getEngineVersionDetails()['MajorVersion'] >= 5:
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 632, in _getEngineVersionDetails
    versionFile = os.path.join(self.getEngineRoot(), 'Engine', 'Build', 'Build.version')
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 60, in getEngineRoot
    self._engineRoot = self._getEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 620, in _getEngineRoot
    return self._detectEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerLinux.py", line 15, in _detectEngineRoot
    if self._getEngineVersionDetails()['MajorVersion'] >= 5:
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 632, in _getEngineVersionDetails
    versionFile = os.path.join(self.getEngineRoot(), 'Engine', 'Build', 'Build.version')
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 60, in getEngineRoot
    self._engineRoot = self._getEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 620, in _getEngineRoot
    return self._detectEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerLinux.py", line 15, in _detectEngineRoot
    if self._getEngineVersionDetails()['MajorVersion'] >= 5:
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 632, in _getEngineVersionDetails
    versionFile = os.path.join(self.getEngineRoot(), 'Engine', 'Build', 'Build.version')
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 60, in getEngineRoot
    self._engineRoot = self._getEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 620, in _getEngineRoot
    return self._detectEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerLinux.py", line 15, in _detectEngineRoot
    if self._getEngineVersionDetails()['MajorVersion'] >= 5:
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 632, in _getEngineVersionDetails
    versionFile = os.path.join(self.getEngineRoot(), 'Engine', 'Build', 'Build.version')
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 60, in getEngineRoot
    self._engineRoot = self._getEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 620, in _getEngineRoot
    return self._detectEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerLinux.py", line 15, in _detectEngineRoot
    if self._getEngineVersionDetails()['MajorVersion'] >= 5:
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 632, in _getEngineVersionDetails
    versionFile = os.path.join(self.getEngineRoot(), 'Engine', 'Build', 'Build.version')
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 60, in getEngineRoot
    self._engineRoot = self._getEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 620, in _getEngineRoot
    return self._detectEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerLinux.py", line 15, in _detectEngineRoot
    if self._getEngineVersionDetails()['MajorVersion'] >= 5:
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 632, in _getEngineVersionDetails
    versionFile = os.path.join(self.getEngineRoot(), 'Engine', 'Build', 'Build.version')
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 60, in getEngineRoot
    self._engineRoot = self._getEngineRoot()
  File "/home/russ/work/ue4cli/ue4cli/UnrealManagerBase.py", line 615, in _getEngineRoot
    override = ConfigurationManager.getConfigKey('rootDirOverride')
  File "/home/russ/work/ue4cli/ue4cli/ConfigurationManager.py", line 30, in getConfigKey
    configFile = ConfigurationManager._configFile()
  File "/home/russ/work/ue4cli/ue4cli/ConfigurationManager.py", line 45, in _configFile
    return os.path.join(ConfigurationManager.getConfigDirectory(), 'config.json')
  File "/home/russ/work/ue4cli/ue4cli/ConfigurationManager.py", line 16, in getConfigDirectory
    if 'UE4CLI_CONFIG_DIR' in os.environ:
  File "/usr/lib/python3.10/_collections_abc.py", line 825, in __contains__
    self[key]
  File "/usr/lib/python3.10/os.py", line 676, in __getitem__
    value = self._data[self.encodekey(key)]
RecursionError: maximum recursion depth exceeded
```

**Comment by kreempuff** [comment]
To add some color here, I experienced this error while using ue4cli in the official Unreal Engine 5 build container.

**Comment by russkel** [comment]
Oops, added some text about the CLI version and engine version.

**Comment by kirannhegde** [comment]
Hello @russkel @kreempuff   I am running into the same issue now. Any pointers on how this was resolved?

I run into this error with any command that i execute such as `ue4 root`
Any help would be very much appreciated.

Thanks,
Kiran Hegde

**Comment by rhighs** [comment]
I'm experiencing the same issue on `ue4 build-target UE4Editor Development`

Unreal Engine version 4.27 on Windows.
Installation Path: C:\Program Files\Epic Games\UE_4.27

**Comment by rhighs** [comment]
Hi everyone, I was able to solve the issue by setting manually the unreal engine root using `ue4 setroot <UNREAL_ENGINE_ROOT_DIR>`. If you are wondering what ue4 considers as root it should be the one named as `UnrealEngine`, if that is not the case, look for the directory containing the `UE4.sln` file.

**Comment by adamrehn** [comment]
This is fixed by the merge of #58.
